tests/unit/test_level_based_platform_query_engine.py

```diff
@@ -17,67 +18,47 @@ def setUp(self):
         self.level2_key = "thread"
         self.platform_table_name = "discord"
         self.date_key = "date"
-        self.engine = LevelBasedPlatformQueryEngine(
-            level1_key=self.level1_key,
-            level2_key=self.level2_key,
-            platform_table_name=self.platform_table_name,
-            date_key=self.date_key,
-        )
+        os.environ["OPENAI_API_KEY"] = "sk-some_creds"
 
     def test_prepare_platform_engine(self):
         """
         Test prepare_platform_engine method with sample data
         """
-        level1_names = ["general"]
-        level2_names = ["discussion"]
-        days = ["2022-01-01"]
-        query_engine = self.engine.prepare_platform_engine(
+        # the output should always have a `date` key for each dictionary
+        filters = [
+            {"channel": "general", "date": "2023-01-02"},
```

Comment:
Do we have test cases where multiple channels/threads/dates are provided?

Reply:
This file contains test cases for applying multiple `OR` between each item of the filters list and two or one `AND` operation between each item of the list (each dictionaries). Adding a `thread` to it, is no different as it would add three `AND` operator instead of two for date and channel.